pkg/check/util.go

```diff
@@ -227,3 +254,60 @@ func vCentersToMap(vcenters []ocpv1.VSpherePlatformVCenterSpec) map[string]ocpv1
 	}
 	return vcenterMap
 }
+
+func GetVCenter(checkContext *CheckContext, node *v1.Node) (*VCenter, error) {
+	if checkContext.PlatformSpec != nil {
+		// Determine which vCenter connection is in by looking at failure domain info.  If topology label is not found,
+		// fall back and look for older beta version.
+		region := node.Labels[v1.LabelTopologyRegion]
+		if len(region) == 0 {
+			region = node.Labels[v1.LabelFailureDomainBetaRegion]
+		}
+		zone := node.Labels[v1.LabelTopologyZone]
+		if len(zone) == 0 {
+			zone = node.Labels[v1.LabelFailureDomainBetaZone]
+		}
+
+		// In older clusters, the region/zone will be blank on nodes and there will be no FD.
+		if len(region) > 0 && len(zone) > 0 {
+			klog.V(2).Infof("Checking for region %v zone %v", region, zone)
+			server := ""
+			// Get failure domain
+			for _, fd := range checkContext.PlatformSpec.FailureDomains {
+				if fd.Region == region && fd.Zone == zone {
+					server = fd.Server
+					break
+				}
+			}
+
+			// Maybe return error if server not found?
+			if server == "" {
+				return nil, fmt.Errorf("unable to determine vcenter for node %s", node.Name)
+			}
+
+			return checkContext.VCenters[server], nil
+		} else {
+			// Node is not configured w/ FD labels.  Admin may not have updated control plane after migrating to
+			// multi vCenter config.
+			klog.Warningf("failure domain labels missing from node %s", node.Name)
+			if len(checkContext.PlatformSpec.FailureDomains) > 0 {
+				klog.V(2).Infof("Returning first FD defined in platform spec.")
+				for _, fd := range checkContext.PlatformSpec.FailureDomains {
+					return checkContext.VCenters[fd.Server], nil
+				}
+			} else {
+				klog.V(2).Infof("There are no failure domains in platform spec. Returning first vCenter from context.")
+				for index := range checkContext.VCenters {
+					return checkContext.VCenters[index], nil
+				}
+			}
+		}
+	} else {
+		// Platform spec is not set.  This is old behavior before FDs.  Return only FD.
```

Comment:
Is it possible to be here for multi vCenter case? If not, we could assert or return err if len(checkContext.VCenters)>1. If yes, let's add `klog.V(2).Infof()` print here saying that we return first vCenter from the list.

Reply:
For multi vCenter, we should never get in here; however, I do like your suggestion of adding the logging message so I'll add that to this block to help future debug.  I'll also add the vcenters > 1 err logic.